docs/proposals/drafts/2022023-remote-write-observatorium-api.md

```diff
@@ -0,0 +1,57 @@
+---
+toc: true
+title: 2022-02 Additional Metrics Write Endpoints
+---
+
+# Additional Metrics Write Endpoints Proposal
+
+* **Owners:**:
+  * `@marcolan018`
+
+* **Related Tickets:**
+  * `<JIRA, GH Issues>`
+
+* **Other docs:**
+  * [Exporting Metrics To Additional Consumers in RedHat Advanced Cluster Management for Kubernetes](https://docs.google.com/document/d/1t7GodF4s_V5MtFtjTeiT0UFeQuu7cuSzz9-7FEENDsQ) (internal Red Hat link)
+
+## TLDR
+
+We propose to support additional metrics write endpoints. The metrics write requests can be forwarded to additional backend endpoints, which support prometheus remote write protocol, besides thanos receivers.
+
+## Why
+Currently the observatorium API always forwards the metrics to Thanos. Given that users have multiple consumers of metric data, they have a requirement for the collected metrics to be made available to these additional consumers. Although the users can pull the metrics from Thanos side directly, but that's not a realtime solution. Also it will lead to heavy workload on Thanos if users consistently query massive metrics from it. We need a way to let observatorium API to forward the metrics to more than one targets.
+
+## Goals
+
+* Observatorium API can pushes data to configured additional metrics write endpoints, secured or non-secured
+* Observatorium API continues to push data to Thanos Receivers even when the additional metrics write endpoints blocked.
+
+## Non-Goals
+
+* Other types of observability data(e.g. logs) to multiple endpoints not considered
+
+## How
+
+Currently Observatorium  API only supports one metrics write endpoint. We propose an update in Observatorium API, to add a new component RemoteWrite Proxy, to clone the request body of incoming metrics write requests, then forward to additional write endpoints, which support prometheus remote write protocol.
```

Comment:
I think we don't need to go too much into implementation details, but could it be clarified here what is RemoteWrite Proxy? Will it be a true new stand-alone component or this will be merely internal part of the API service?

Reply:
change the term "component" to "handler" to avoid confusing